### ISSUE TITLE
Handling of multiple watchdirs is undefined

### ISSUE THREAD

**Issue by e2** [guard/guard#634]
#### TL;DR - avoid using multiple watched directories for now (-w option). If possible, watch the entire project and use "smarter" rules or custom plugins.

Example:

Given a tree:

```
/project/Guardfile
/project/spec/models/foo_spec.rb
/project/moduleA/spec/models/foo_spec.rb
```

watching `/project` and `/project/modules`, e.g.

```
guard -w . moduleA
```

results in ambiguous or non-existent relative paths passed to plugins, e.g.

```
touch /project/moduleA/spec/models/foo_spec.rb
```

triggers the event:

``` ruby
:modified, spec/models/foo_spec.rb  # because the /project/moduleA was substituted
```
### Problem

Working with relative paths is more convenient (matching regexps in Guardfile), although the concept of multiple watch dirs breaks this.
### Workaround

Reorganizing projects and directories to avoid needing multiple watched dirs with custom rules/configurations (consider multiple Guardfiles rather than multiple watch dirs).
### Proper solution

Probably a `directory` keyword (similar to the idea of the existing `group` keyword) could allow having different plugin configurations for different directories.
### Alternative solution

All paths and regexps should be either assumed relative to the given working directory, or absolute (if outside the project).
